src/server/cluster_config_test.cc

```diff
@@ -126,6 +126,51 @@ TEST_F(ClusterConfigTest, ConfigSetMultipleInstances) {
   }
 }
 
+TEST_F(ClusterConfigTest, ConfigSetGetDeletedSlots) {
+  auto ds =
+      config_.SetConfig({{.slot_ranges = {{.start = 0, .end = 5'000}},
+                          .master = {.id = kMyId, .ip = "192.168.0.100", .port = 7000},
+                          .replicas = {}},
+                         {.slot_ranges = {{.start = 5'001, .end = 10'000}},
+                          .master = {.id = "other-master", .ip = "192.168.0.102", .port = 7002},
+                          .replicas = {}},
+                         {.slot_ranges = {{.start = 10'001, .end = 0x3FFF}},
+                          .master = {.id = "other-master2", .ip = "192.168.0.104", .port = 7004},
+                          .replicas = {}}});
+
+  EXPECT_TRUE(ds);
```

Comment:
Nit: a matter of preference, but I'd use `ds.has_value()` over implicit bool conversion (also below)